src/qibo/quantum_info/basis.py

```diff
@@ -143,33 +157,50 @@ def comp_basis_to_pauli(
             performed row-wise. If ``"column"``, vectorization is performed
             column-wise. If ``"system"``, system-wise vectorization is
             performed. Default is ``"row"``.
+        backend (``qibo.backends.abstract.Backend``, optional): backend to be
+            used in the execution. If ``None``, it uses ``GlobalBackend()``.
+            Defaults to ``None``.
 
     Returns:
         ndarray or tuple: Unitary matrix :math:`U`. If ``sparse=True``,
             tuple is composed of array of non-zero elements and an
             array with their row-wise indexes.
 
     """
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
 
     if sparse:
         elements, indexes = pauli_basis(
-            nqubits, normalize, vectorize=True, sparse=sparse, order=order
+            nqubits,
+            normalize,
+            vectorize=True,
+            sparse=sparse,
+            order=order,
+            backend=backend,
         )
         elements = np.conj(elements)
 
         return elements, indexes
 
     unitary = pauli_basis(
-        nqubits, normalize, vectorize=True, sparse=sparse, order=order
+        nqubits, normalize, vectorize=True, sparse=sparse, order=order, backend=backend
     )
 
     unitary = np.conj(unitary)
 
+    if backend.__class__.__name__ == "TensorflowBackend":
+        unitary = backend.cast(unitary, dtype=unitary.dtype)
```

Comment:
If you do this `cast` for all backends, so that you don't need to check explicitly if it is tensorflow, does it break something?

Reply:
It doesn't break but it's redundant. I opted to check if because for bigger systems it might be a slow process to recast to the same backend even though the array is already at the right backend.

Reply:
Changed where arrays are cast to the general backend, and now this is working